### PR TITLE
fix: Text.set_color_by_t2c and Text.set_color_by_t2g little changes

### PR DESCRIPTION
**Pull request by LedMeExplain**
## Changelog / Overview
There was a little bug at t2c and t2g Text functions when using it at text with one ore more blank spaces.

## Explanation for Changes
Just changed self.original_text for self.text since 'original_text' contains blank spaces that are not used after init.

### Example
```py
class T2cT2g(Scene):
    def construct(self):
        t1 = Text("Hi there, this is a t2c, t2g  fix",
            t2c = {"Hi":RED,"this":GREEN,"t2c":BLUE},
            t2g = {"there":(RED,BLUE)}
        )
        self.add(t1)
```
#### Original t2c and t2g
image
#### Fixed t2c and t2g
image

## Checklist
- [ ] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [ ] I have written a descriptive PR title (see top of PR template for examples)
- [ ] I have written a changelog entry for the PR or deem it unnecessary
- [ ] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by naveen521kk** [comment]
Can you confirm this doesn't break when using `disable_ligature` parameter with Text?

**Comment by LedMeExplain** [comment]
> Can you confirm this doesn't break when using `disable_ligature` parameter with Text?

You were so right. I forgot to test in using `disable_ligatures`. I added a condition for using `text` or `original_text` depending on `disable_ligatures`.

### Test
```py
class T2cT2gDL(Scene):
    def construct(self):

        li = Text("fl ligature",
            t2c = {"f":RED,"li":GREEN,"e":BLUE},
            t2g = {"atu":(RED,BLUE)}
        ).scale(2)

        nli = Text("fl ligature", 
            disable_ligatures=True,
            t2c = {"f":RED,"li":GREEN,"e":BLUE},
            t2g = {"atu":(RED,BLUE)}
        ).scale(2)

        self.add(Group(li, nli).arrange(DOWN, buff=.8))
```
#### Current Result
image

#### Fixed Result
image

**Comment by huguesdevimeux** [comment]
Please add tests :)

File: manim/mobject/types/vectorized_mobject.py [hunk 1]
```diff
@@ -49,6 +49,8 @@
 
 
 class VMobject(Mobject):
+    """A vectorized mobject."""
```
Comment (naveen521kk) [inline manim/mobject/types/vectorized_mobject.py:52]: Can you do this in a new PR?

Comment (LedMeExplain) [reply manim/mobject/types/vectorized_mobject.py]: I'm sorry, actually it is in another PR but I made a mistake and added that code here too. Here's the PR with these changes #1749
I just deleted these changes in this PR

**Comment by LedMeExplain** [comment]
> Please add tests :)

@huguesdevimeux I'm not sure how to do this. I'm new at contributing but I'll take ss to the pytest results.

image
image
image
image
image
image

**Comment by naveen521kk** [comment]
I think @huguesdevimeux meant to create a few tests related to your changes as in create an example `Text` object and check whether the Mobject are actually coloured correctly as expected. That would make sure it doesn't break later.

**Comment by LedMeExplain** [comment]
> I think @huguesdevimeux meant to create a few tests related to your changes as in create an example `Text` object and check whether the Mobject are actually coloured correctly as expected. That would make sure it doesn't break later.
> 
> 

Oh, ok. I'll be working on that 👍

**Comment by behackl** [comment]
Just to clarify a bit further: I think we can't add graphical unit tests for `Text` due to OS-dependent fonts and text rendering, but a "normal" unit test probably works. You could add a file `test_text.py` in the `tests` directory that contains a function 

```py
def test_t2c():
    t = Text("fl ligature", t2c=...)
    assert ...
    assert ...
```

where in the assertion part coloring of the glyphs (accessed as t[0], t[1], ...) should be checked.

**Comment by LedMeExplain** [comment]
> Just to clarify a bit further: I think we can't add graphical unit tests for `Text` due to OS-dependent fonts and text rendering, but a "normal" unit test probably works. You could add a file `test_text.py` in the `tests` directory that contains a function
> 
> ```python
> def test_t2c():
>     t = Text("fl ligature", t2c=...)
>     assert ...
>     assert ...
> ```
> 
> where in the assertion part coloring of the glyphs (accessed as t[0], t[1], ...) should be checked.

@behackl ok, I'll do it.

By the way, I tried with the multi-language example and found out another little bug while trying to color a string with blank spaces so i fixed it too. Now you can specify any string in `t2c` or `t2g` exactly as it is originally. I'm working on Windows 10 so the only way of displaying `morning`, `hin` and `arb` was to set the font to `Nirmala UI`

```py
class T2cTest(Scene):
    def construct(self):
        lig = True
        ver = "Original"
        title = Text(ver+r" version, disable_ligatures = "+str(lig))

        morning = Text(
            "வணக்கம்",
            t2c={"வ": BLUE,"ம": RED},
            t2g={"ணக்க": (TEAL,PURPLE)},
            disable_ligatures = lig,
            font="Nirmala UI"
        )

        chin = Text(
            "見 角 言 谷  辛 辰 辵 邑 酉 釆 里!",
            t2c={"見": BLUE,"!": RED},
            t2g={"言 谷  辛 辰": (TEAL,PURPLE)},
            disable_ligatures = lig,
            font="sans-serif"
        )

        mess = Text(
            "Multi-Language",
            weight=BOLD,
            t2c={"M": BLUE,"e": RED},
            t2g={"i-Lang": (TEAL,PURPLE)},
            disable_ligatures = lig,
            font="sans-serif"
        )

        russ = Text(
            "Здравствуйте मस नम म",
            t2c={"З": BLUE,"म": RED},
            t2g={"вуйте मस": (TEAL,PURPLE)},
            disable_ligatures = lig,
            font="Nirmala UI",
        )

        hin = Text(
            "नमस्ते",
            t2c={"न": BLUE,"स्ते": RED},
            disable_ligatures = False, #Due to a gen_chars problem
            font="Nirmala UI",
        )

        arb = Text(
            "صباح الخير تشرفت بمقابلتك",
            t2c={"ر": BLUE,"ت": RED},
            disable_ligatures = lig,
            font="Nirmala UI",
        )

        esp = Text(
            "Texto en \n varias lineas",
            t2c={"T": BLUE,"s": RED},
            t2g={"en \n varias": (TEAL,PURPLE)},
            disable_ligatures = lig,
            font="sans-serif",
        )

        japanese = Text(
            "臂猿「黛比」帶著孩子",
            t2c={"臂": BLUE,"子": RED},
            t2g={"「黛比」帶": (TEAL,PURPLE)},
            disable_ligatures = lig,
            font="Nirmala UI",
        )

        group = Group(title,morning,chin,mess,russ,hin,arb,esp,japanese)
        group.arrange(DOWN).scale_to_fit_height(config['frame_height']-1)
        self.add(group)
```

I don't really know anything about arabic language so I had no idea how to color it correctly. We'll need an arabic person to help us testing that string.

While testing `disable_ligatures = true`, hin text crashes because of the function `gen_chars` (It has nothing to do with my changes, it is another kind of problem. I checked the function but I’m not sure how to fix this, maybe report it as a bug?)
image

### Results
#### Original code
image
image
#### Fixed code
image
image

**Comment by LedMeExplain** [comment]
@behackl I added the `test_text.py` file and it seems to be working fine.

**Review by behackl — CHANGES_REQUESTED** [review]
Tests are failing with

```
FAILED tests/test_text.py::test_t2c - IndexError: list index out of range
```

**Comment by LedMeExplain** [comment]
@behackl The problem was that I'm working on Windows. So letters 'fl' don't create a ligature like they do on Linux and Mac OS. I simply change the indexes to check the first and last position of the Text

**Comment by LedMeExplain** [comment]
It's not working for linux and mac os :( 
Guess my code only fix the problem on windows. Maybe it's not worth it to keep going with this branch. What do you guys think?
I figured out that t2c and t2g colors everything correctly if disable_ligatures is set to False. It could be a temporary solution. Guess it would be better to put some examples at docs specifying this

**Pull request closed without merging**